packages/next/src/build/swc/index.ts

```diff
@@ -1150,14 +1185,33 @@ async function loadWasm(importPath = '') {
         minifySync(src: string, options: any) {
           return bindings.minifySync(src.toString(), options)
         },
-        parse(src: string, options: any) {
-          return bindings?.parse
-            ? bindings.parse(src.toString(), options)
-            : Promise.resolve(bindings.parseSync(src.toString(), options))
-        },
-        parseSync(src: string, options: any) {
-          const astStr = bindings.parseSync(src.toString(), options)
-          return astStr
+        analysis: {
+          isDynamicMetadataRoute: async (pageFilePath: string) => {
+            const fileContent = (await tryToReadFile(pageFilePath, true)) || ''
```

Comment:
For swc-loader transforms we have the SWC-side read the file, seems we could do the same here in a follow-up PR.